File: LeetCode 75 Study Plan to Ace Interviews/Q1480 - Running Sum of 1d Array/solution.py
# solution 1: create a new array for sums
class Solution1:
    def runningSum(self, nums: List[int]) -> List[int]:
        sums = []
        newSum = 0
        for num in nums:
            newSum += num
            sums.append(newSum)
        return sums


# solution 2: modify nums directly
class Solution2:
    def runningSum(self, nums: List[int]) -> List[int]:
        for index in range(1, len(nums)):
            nums[index] += nums[index - 1]
        return nums

File: LeetCode 75 Study Plan to Ace Interviews/Q1480 - Running Sum of 1d Array/solution.cs
// solution 1: create a new array for sums
public class Solution {
    public int[] RunningSum(int[] nums) {
        int newSum = 0;
        int[] sums = new int[nums.Length];
        for (int i = 0; i < nums.Length; i++){
            newSum += nums[i];
            sums[i] = newSum;
        };
        return sums;
    }
}

// solution 2: modify nums directly
public class Solution2 {
    public int[] RunningSum(int[] nums) {
        for(int i = 1; i < nums.Length; i++){
            nums[i] = nums[i - 1] + nums[i];
        }
        return nums;
    }
}